src/App.tsx

```diff
@@ -14,11 +15,99 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Selected good: -</h1>
-    {goodsFromServer.length}
-  </div>
-);
+type State = {
+  selectedGood: string[],
+};
+
+class App extends React.Component<{}, State> {
+  state: State = {
+    selectedGood: ['Jam'],
+  };
+
+  addSelectedGood = (good: string) => {
+    this.setState((prevState) => (
+      { selectedGood: [...prevState.selectedGood, good] }
+    ));
+  };
+
+  removeSelectedGood = (good: string) => {
+    this.setState((prevState) => ({
+      selectedGood: prevState.selectedGood.filter((item) => item !== good),
+    }));
+  };
+
+  createMessage = (goods: string[]) => {
+    switch (goods.length) {
+      case 0:
+        return 'No goods selected';
+      case 1:
+        return `${goods[0]} is selected`;
+      case 2:
+        return `${goods.join(' and ')} are selected`;
+      default:
+        return `${goods.slice(0, -1).join(', ')} and ${goods[goods.length - 1]} are selected`;
+    }
+  };
+
+  clearGoods = () => (
+    this.setState({ selectedGood: [] })
+  );
+
+  render() {
+    const noGoodsSelected: boolean = this.state.selectedGood.length === 0;
+    const clearButtonClasses = classNames(
+      {
+        clearButtonHidden: noGoodsSelected,
+      },
+    );
+
+    return (
+      <div className="App">
+        <div>
+          <h1 className="title">
+            {this.createMessage(this.state.selectedGood)}
+            { ' ' }
+
+            <button
+              type="button"
+              className={clearButtonClasses}
+              onClick={this.clearGoods}
+            >
+              Clear
+            </button>
```

Comment:
not sure why this button has to be inside `h1`

Reply:
right i will remove it